src/clusterfuzz/_internal/bot/fuzzers/syzkaller/runner.py

```diff
@@ -181,7 +184,7 @@ def repro(self, repro_timeout, repro_args):
     logs.log('Running Syzkaller testcase.')
```

Comment:
Can we add the retry here instead? 

i.e. if `lost connection to test machine` is in the logs, retry until we hit the retry limit or it's no longer in the logs. 

The `IGNORE_REPRODUCED_CRASH_TYPE` logic is very syzkaller/android specific, and it's better to keep it closer to here.

Reply:
>better to keep it closer to here
I agree that this issue is very syzkaller specific. What I'm afraid is changes here won't actually resolve the reliably reproducible flag issue. Seems like this logic of marking whether the crash reproduces is independent of the `reliably reproducible` flag that gets set by a job running `test_reproduce_reliability`?

Reply:
`runner.repro()` seems to currently only check whether the repro job is done and whether the crash happened or not (even if the crash may be of a wrong type)

Reply:
> Seems like this logic of marking whether the crash reproduces is independent of the reliably reproducible flag that gets set by a job running test_reproduce_reliability?

It should end up with the same outcome if we do it here, unless I'm misunderstanding your question. `test_reproduce_reliability` etc will all end up calling this code. It's also better to fix this here so that it generalises to anywhere else that tries to reproduce a syzkaller testcase. 

What I'm suggesting is to add logic to `runner.repro` to retry if the "lost connection" message is in the log somewhere.

Reply:
ohh I think I understand where the gap is.

I don't know about `libFuzzer` or other fuzzers, but for Syzkaller it has its own tool called `syz-crush` for reproducing. `runner.repro()` uses `syz-crush` which outputs `syz-crush` log which is different than the log `syzkaller` originally produced. I am curious how clusterfuzz reads crash state from syz-crush output because `syz-crush` log does not provide any call trace.

Reply:
I found where `syz-crush` stashes its logs. It does produce an output log like the original one including call trace (crash state). This current implementation doesn't actually open this log and uses `syz-crush` output. I'll make the appropriate change.

Reply:
opening a new PR.